Add user list fetch interface

AaricChenLegacy opened the pull request:
增加了获取用户列表的接口，之前没有，希望merge

Commit:
Add user list fetch interface

ihaolin commented on src/main/java/me/hao0/wechat/core/Users.java:
```diff
@@ -421,34 +458,76 @@ public User getUser(String accessToken, String openId){
+    public List<String> getUserList(String accessToken, String nextOpenId) {
+        checkNotNullAndEmpty(accessToken, "accessToken");
+        String url;
+        if (nextOpenId == null || nextOpenId.equalsIgnoreCase("")) {
```
这里这样写，是不是更好一些：

```
String url = GET_USER_LIST_INFO + accessToken;
if (!Strings.isNullOrEmpty(nextOpenId)){
    url += ("&next_openid=" + nextOpenId); 
}
```

ihaolin commented on src/main/java/me/hao0/wechat/core/Users.java:
```diff
@@ -421,34 +458,76 @@ public User getUser(String accessToken, String openId){
+     * @param nextOpenId  nextOpenId
+     * @return 用户列表，或抛WechatExeption
+     */
+    public List<String> getUserList(String accessToken, String nextOpenId) {
```
建议将**getUserList**命名为**getUsers**

ihaolin commented on src/main/java/me/hao0/wechat/core/Users.java:
```diff
@@ -421,34 +458,76 @@ public User getUser(String accessToken, String openId){
+        }
+        Map<String, Object> resp = doGet(url);
+
+        List<String> openIds = (List<String>) ((Map) resp.get("data")).get("openid");
```
官方文档，返回的数据应该还有**total**，**count**，**next_openid**等信息，建议将起封装为对象供外部使用。

ihaolin replied:
请记得在加上**单元测试**。

Commit:
Users info done
 - change getUserList() to getUsers()
 - add getUsers() result entity UserList
 - add test
